app/uk/gov/hmrc/gform/service/RepeatingComponentService.scala

```diff
@@ -121,29 +125,29 @@ class RepeatingComponentService @Inject() (val sessionCache: SessionCacheConnect
     }
   }
 
-  private def evaluateExpression(expr: Expr, formTemplate: FormTemplate, data: Map[FieldId, Seq[String]])(implicit hc: HeaderCarrier): Int = {
+  private def evaluateExpression(expr: Expr, formTemplate: FormTemplate, data: Map[FieldId, Seq[String]])(implicit hc: HeaderCarrier, ex: ExecutionContext): Future[Int] = {
     expr match {
-      case Add(expr1, expr2) => evaluateExpression(expr1, formTemplate, data) + evaluateExpression(expr2, formTemplate, data)
-      case Multiply(expr1, expr2) => evaluateExpression(expr1, formTemplate, data) * evaluateExpression(expr2, formTemplate, data)
+      case Add(expr1, expr2) => evaluateExpression(expr1, formTemplate, data).flatMap(x => evaluateExpression(expr2, formTemplate, data).map(_ + x))
+      case Multiply(expr1, expr2) => evaluateExpression(expr1, formTemplate, data).flatMap(x => evaluateExpression(expr2, formTemplate, data).map(_ * x))
       case Sum(FormCtx(expr1)) =>
         val dataGetter: FieldId => Int = fieldId => Try(data.get(fieldId).toList.flatten.headOption.getOrElse("0").toInt).getOrElse(0)
         val cacheMap: Future[CacheMap] = getAllRepeatingGroups
-        val repeatingSections = formTemplate.sections.flatMap(_.fields).map(fv => (fv.id, fv.`type`)).collect {
-          case (fieldId, group: Group) => Await.result(cacheMap.map(_.getEntry[List[List[FieldValue]]](fieldId.value).getOrElse(Nil)), 10 seconds)
-        }
-        Group.getGroup(repeatingSections, FieldId(expr1)).map(dataGetter).sum
+        val repeatingSections: Future[List[List[List[FieldValue]]]] = Future.sequence(formTemplate.sections.flatMap(_.fields).map(fv => (fv.id, fv.`type`)).collect {
+          case (fieldId, group: Group) => cacheMap.map(_.getEntry[List[List[FieldValue]]](fieldId.value).getOrElse(Nil))
+        })
+        Group.getGroup(repeatingSections, FieldId(expr1)).flatMap(x => Future.successful(x.map(dataGetter).sum))
       case formExpr @ FormCtx(_) => getFormFieldIntValue(TextExpression(formExpr), data)
       case Constant(value) => Try(value.toInt) match {
-        case Success(intValue) => intValue
-        case _ => 0
+        case Success(intValue) => Future.successful(intValue)
+        case _ => Future.successful(0)
       }
       //      case AuthCtx(value: AuthInfo) =>
       //      case EeittCtx(value: Eeitt) =>
-      case _ => 0
+      case _ => Future.successful(0)
     }
   }
 
-  private def getRequestedCount(expr: TextExpression, formTemplate: FormTemplate, data: Map[FieldId, Seq[String]], cacheMap: CacheMap)(implicit hc: HeaderCarrier) = {
+  private def getRequestedCount(expr: TextExpression, formTemplate: FormTemplate, data: Map[FieldId, Seq[String]], cacheMap: CacheMap)(implicit hc: HeaderCarrier): Future[Int] = {
```

Comment:
I don;t think this needs to return a future, I think it is better to wrap this function's result in a Future.successful where it is used